tests/roles/ovn_adoption/tasks/main.yaml

```diff
@@ -29,31 +29,77 @@
   retries: 60
   delay: 2
 
-- name: get podified OVN NB ovsdb-server IP
+- name: get podified OVN NB ovsdb-server service cluster IP
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc get pod ovsdbserver-nb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status}' | \
-    jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | \
-    tr -d '"'
+    oc get svc --selector "statefulset.kubernetes.io/pod-name=ovsdbserver-nb-0" -ojsonpath='{.items[0].spec.clusterIP}'
   register: podified_ovn_nb_ip_result
 
 - name: get podified OVN SB ovsdb-server IP
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc get pod ovsdbserver-sb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status}' | \
-    jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | \
-    tr -d '"'
+    oc get svc --selector "statefulset.kubernetes.io/pod-name=ovsdbserver-sb-0" -ojsonpath='{.items[0].spec.clusterIP}'
   register: podified_ovn_sb_ip_result
 
+- name: set fact for oc run args
+  ansible.builtin.set_fact:
+    oc_run_args:
+      spec:
+        containers:
+        - image: "{{ ovn_image }}"
+          command: [ "sh", "-c", "sleep infinity"]
+          name: adoption
+          volumeMounts:
+          - mountPath: /backup
+            name: backup
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop: ALL
+          runAsNonRoot: true
+          runAsUser: 1000
+          seccompProfile:
+            type: RuntimeDefault
+        volumes:
+        - hostPath:
+            path: "{{ playbook_dir }}/{{ ovn_copy_tmp_dir }}"
+            type: Directory
+          name: backup
+
+- name: prepare oc run args file
+  copy:
+    content: "{{ oc_run_args }}"
+    dest: "{{ ovn_copy_tmp_dir }}/oc_run_args.yaml"
+  register: oc_run_args_file
+
+- name: set fact for oc run command
+  ansible.builtin.set_fact:
+    oc_run_cmd: >-
+      oc delete pod adoption --ignore-not-found=true;
+      oc run adoption
+      --image {{ ovn_image }}
+      --overrides='$(cat {{ playbook_dir }}/{{ ovn_copy_tmp_dir }}/oc_run_args.yaml)';
+      oc wait --for=condition=Ready pod adoption --timeout=30s
+
+- name: start an adoption helper pod
```

Comment:
Shouldn't it be removed once the adoption had been completed?

Reply:
it should be, right, and there is a handler to do so. But the teardown needs improvement (the pod remains if playbooks exits with errors)

Reply:
so I've added that pre-start command to remove the helper pod left behind in such a failure scenario. That looks simpler than changing all the sections with rescue blocks to remove it. We can think on a better solution

Reply:
Thanks! After looking a bit I discovered that you can add "force_handlers: true" to a task to ensure that even if there were some error during the play the handler will run. 

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#handlers-and-failure

Just sharing it, I like the current implementation :) Thanks!